mitodl/mitxonline
# feat: refund orders CyberSource - Integrate [mitol-django-payment-gateway]

#### Pre-Flight checklist


#### What are the relevant tickets?
#571 

#### What's this PR do?
- Adds the initial integration of order refunds with the `mitol-django-payment-gateway` from https://github.com/mitodl/ol-django/pull/64.
- Removes all the Cybersource related settings from MITxOnline Marketing site (They have been migrated to the common app ttps://github.com/mitodl/ol-django/pull/64)
- Implements the refund flow using API and adds integration for refunds done through Django Admin
- Also, updates other ol-django packages to for compatibility reasons

#### How should this be manually tested?
- Setup the all the environment-related config for CyberSource, (You can see them all [here](https://github.com/mitodl/ol-django/pull/64/files#diff-8925307c256c81211d35d1a950207fb606d513e45d72cc6ab6ccaa4c44f5d431))
- Setup a product with a price and place an order for it (Can be done while enrolling in a course or by `Get Certificate` links on the course details and dashboard pages) - You should complete all Cybersource steps successfully for placing an order
- You can test it through Django Shell or Admin. The full flow should be testable as:
    - Place an order
    - Open the Fulfilled order from above step in Django Admin and click the `Refund Order` button
    - The Order should be refunded, and the user should be unenrolled of the checkbox was selected while refunding in the above step
    - Notice that a refund Transaction with Cyber Source API response details gets created and the order status is changed to `Refunded`

#### Where should the reviewer start?
Setup, product, place order, complete CyberSource order steps successfully before a refund

#### NOTE:
- If you don't have the configuration keys for CyberSource you can ask me or a fellow developer

Below is Done:
- ~This PR is associated with its counterpart work done in ol-django in https://github.com/mitodl/ol-django/pull/64.~
- ~This PR should ideally be reviewed after https://github.com/mitodl/ol-django/pull/64~
